File: HoTT-Model/src/main/java/de/treichels/hott/model/voice/CountryCode.java
package de.treichels.hott.model.voice;

import java.util.ResourceBundle;
import java.util.stream.Stream;

public enum CountryCode {
    GLOBAL(0), US(1), KR(2), CN(3), JP(4);

    public static CountryCode forCode(final int countryCode) {
        return Stream.of(CountryCode.values()).filter(c -> c.getCode() == countryCode).findFirst().orElse(GLOBAL);
    }

    private final int code;

    CountryCode(final int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    @Override
    public String toString() {
        return ResourceBundle.getBundle(getClass().getName()).getString(name());
    }
}

File: HoTT-Voice/src/main/kotlin/de/treichels/hott/voice/CountryCode.kt
package de.treichels.hott.voice

import de.treichels.hott.util.get
import java.util.*

enum class CountryCode(val code: Int) {
    GLOBAL(0), US(1), KR(2), CN(3), JP(4);

    override fun toString(): String = ResourceBundle.getBundle(javaClass.name)[name]

    companion object {
        @JvmStatic
        fun forCode(countryCode: Int): CountryCode = values().find { it.code == countryCode } ?: GLOBAL
    }
}
